pwiz_tools/Skyline/Controls/Graphs/DetectionPlotData.cs

```diff
@@ -381,6 +415,9 @@ private void CacheData(DataRequest request, int index)
             {
                 try
                 {
+                    if(!_document.IsLoaded)
+                        SetCacheStatus(CacheStatus.idle, "Waiting for the document to load.");
+                    while(!_document.IsLoaded)  Thread.Sleep(100);

```

Comment:
Better to have a listener for changes to the document than to hang on a busy wait.

Reply:
This wait is for the document load being complete and all document data available. It looks like OnDocumentChanged event is different and happens when document is changed in the UI. I do not see any events I could use for this purpose and implementing one without using some kind of polling seems non-trivial.

Reply:
You can still use OnDocumentChanged and then check inside it if document.IsLoaded. Not all changes will mean the document is loaded, but when the document transitions between !IsLoaded and IsLoaded, you will get this event. You do have to remove this busy wait.

Reply:
OK. Since the client already generates a data request on document change I just made the thread terminate if the document is not loaded instead of waiting. I didn't know that changes of the IsLoaded attribute generate OnDocumentChanged events, it was not obvious from the code.